ui/src/components/Map/LeafletMap.vue

```diff
@@ -0,0 +1,130 @@
+<template>
+  <div class="leaflet">
+    <div class="geo-map">
+      <l-map
+        ref="map"
+        v-model:center="center"
+        :max-zoom="19"
+        :min-zoom="2"
+        :zoom="zoom"
+        :zoomAnimation="true"
+        @ready="onLeafletReady"
+      >
+        <template v-if="leafletReady">
+          <l-control-layers />
+          <l-tile-layer
+            v-for="tileProvider in tileProviders"
+            :key="tileProvider.name"
+            :name="tileProvider.name"
+            :visible="tileProvider.visible"
+            :url="tileProvider.url"
+            :attribution="tileProvider.attribution"
+            layer-type="base"
+          />
+          <marker-cluster :options="{ showCoverageOnHover: false, chunkedLoading: true }">
+            <l-marker
+              v-for="(node, index) in interestedNodes"
+              :key="index"
+              :lat-lng="getCoordinateFromNode(node)"
+              add
+            >
+              <l-popup>{{ node.label }}</l-popup>
+            </l-marker>
+            <l-polyline
+              v-for="(coordinatePair, index) in edges"
+              :key="index"
+              :lat-lngs="[coordinatePair[0], coordinatePair[1]]"
+              color="green"
+            />
+          </marker-cluster>
+        </template>
+      </l-map>
+    </div>
+  </div>
+</template>
+<script setup lang ="ts">
+import { computed, ref, nextTick } from "vue";
+import "leaflet/dist/leaflet.css";
+import {
+  LMap,
+  LTileLayer,
+  LMarker,
+  LPopup,
+  LControlLayers,
+  LPolyline,
+} from "@vue-leaflet/vue-leaflet";
+import MarkerCluster from "./MarkerCluster.vue";
+import { Vue } from "vue-class-component";
+import { useStore } from "vuex";
+import { Coordinates } from "@/types";
+let leafletReady = ref(false);
+let leafletObject = ref("");
+let visible = ref(false);
+let map: any = ref();
+const store = useStore();
+let center = computed(() => {
+  const coordinates: Coordinates = store.getters['mapModule/getMapCenter']
+  return [coordinates.latitude, coordinates.longitude];
+})
+let zoom = ref(2);
+let interestedNodes = computed(() => {
+  return store.getters['mapModule/getInterestedNodes'];
+})
+function getCoordinateFromNode(node: any) {
+  let coordinate: string[] = [];
+  coordinate.push(node.assetRecord.latitude);
+  coordinate.push(node.assetRecord.longitude);
+  return coordinate;
+}
+let interestedNodesID = computed(() => {
+  return store.getters['mapModule/getInterestedNodesID'];
+})
+let edges = computed(() => {
+  let ids = interestedNodesID.value;
+  let interestedNodesCoordinateMap = getInterestedNodesCoordinateMap();
```

Comment:
For all `let` and `var` in this file and elsewhere, we can likely switch to `const` to prevent accidental re-assignment.
Only those that are are being reassigned will need `let`. Adding properties to an object or pushing new values to an array does not qualify as re-assigning so we're good to use `const` for those as well.

Reply:
Thanks, I've updated all three vue files.